### PR TITLE
#0: Add UMD container

### PR DESCRIPTION
**Pull request by tt-rkim**

### CODE REVIEW

**Comment by tt-rkim** [comment]
@asrinivasanTT @kluongTT

UMD test image is looking to be around 1.72GB, so smaller than the other metal one.

```
REPOSITORY                                                    TAG                                                       IMAGE ID       CREATED          SIZE
ghcr.io/tt-rkim/sw-hello-world/tt-umd-hello-world             latest-grayskull                                          89ad2c3549d0   5 minutes ago    1.72GB
```

It looks like a lot of this is build stuff.

```
rkim@e12cs07:~/sw-hello-world$ docker history --no-trunc a5eea0114468
IMAGE                                                                     CREATED          CREATED BY                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                        SIZE      COMMENT
sha256:a5eea0114468b0f7a9e6c1b90bb5b9b3308a4204f37c91f6f8f410ffc89e6539   37 minutes ago   CMD ["./run_unit_tests"]                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                          0B        buildkit.dockerfile.v0
<missing>                                                                 37 minutes ago   RUN |1 ARCH_NAME=grayskull /bin/sh -c ln -s build/test/umd/${ARCH_NAME}/unit_tests run_unit_tests && chmod ugo+x run_unit_tests # buildkit                                                                                                                                                                                                                                                                                                                                                                                                                        637kB     buildkit.dockerfile.v0
<missing>                                                                 42 minutes ago   RUN |1 ARCH_NAME=grayskull /bin/sh -c cmake -B build -G Ninja -DTT_UMD_BUILD_TESTS=ON -DBUILD_SHARED_LIBS=OFF && cmake --build build --target umd_tests # buildkit                                                                                                                                                                                                                                                                                                                                                                                                188MB     buildkit.dockerfile.v0
<missing>                                                                 42 minutes ago   ENV ARCH_NAME=grayskull                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                           0B        buildkit.dockerfile.v0
<missing>                                                                 42 minutes ago   ARG ARCH_NAME                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                     0B        buildkit.dockerfile.v0
<missing>                                                                 46 minutes ago   CMD ["tail" "-f" "/dev/null"]                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                     0B        buildkit.dockerfile.v0
<missing>                                                                 46 minutes ago   WORKDIR /usr/tt-umd                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                               0B        buildkit.dockerfile.v0
<missing>                                                                 46 minutes ago   RUN |2 DEBIAN_FRONTEND=noninteractive UBUNTU_VERSION=20.04 /bin/sh -c git clone --recurse-submodules https://github.com/tenstorrent/tt-umd /usr/tt-umd # buildkit                                                                                                                                                                                                                                                                                                                                                                                                 25.4MB    buildkit.dockerfile.v0
<missing>                                                                 49 minutes ago   RUN |2 DEBIAN_FRONTEND=noninteractive UBUNTU_VERSION=20.04 /bin/sh -c apt-get -y update     && apt-get install -y --no-install-recommends     libc++-17-dev     libc++abi-17-dev     && rm -rf /var/lib/apt/lists/* # buildkit                                                                                                                                                                                                                                                                                                                                    15.8MB    buildkit.dockerfile.v0
<missing>                                                                 50 minutes ago   RUN |2 DEBIAN_FRONTEND=noninteractive UBUNTU_VERSION=20.04 /bin/sh -c apt install -y lsb-release wget software-properties-common gnupg && wget https://apt.llvm.org/llvm.sh     && chmod u+x llvm.sh     && ./llvm.sh 17 # buildkit                                                                                                                                                                                                                                                                                                                               996MB     buildkit.dockerfile.v0
<missing>                                                                 51 minutes ago   RUN |2 DEBIAN_FRONTEND=noninteractive UBUNTU_VERSION=20.04 /bin/sh -c apt -y update && apt install -y apt-utils git git-lfs pkg-config cmake=3.22.2-0kitware1ubuntu20.04.1 cmake-data=3.22.2-0kitware1ubuntu20.04.1 curl wget libhwloc-dev ninja-build patchelf # buildkit                                                                                                                                                                                                                                                                                        410MB     buildkit.dockerfile.v0
<missing>                                                                 3 days ago       RUN |2 DEBIAN_FRONTEND=noninteractive UBUNTU_VERSION=20.04 /bin/sh -c apt -y update     && apt install -y --no-install-recommends ca-certificates gpg wget     && wget -O - https://apt.kitware.com/keys/kitware-archive-latest.asc 2>/dev/null | gpg --dearmor - | tee /usr/share/keyrings/kitware-archive-keyring.gpg >/dev/null     && echo 'deb [signed-by=/usr/share/keyrings/kitware-archive-keyring.gpg] https://apt.kitware.com/ubuntu/ focal main' | tee /etc/apt/sources.list.d/kitware.list >/dev/null     && rm -rf /var/lib/apt/lists/* # buildkit   10.6MB    buildkit.dockerfile.v0
<missing>                                                                 3 days ago       ENV CCACHE_TEMPDIR=/tmp/ccache                                                                                                                                                                                                                                                                                                  
```

Could probably get rid of this in a future pass.

File: tt-umd/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+ENV ARCH_NAME=${ARCH_NAME}
+ARG DEBIAN_FRONTEND=noninteractive
+
+# Use a newer version of CMake than what is available from Canonical for 20.04
+RUN apt -y update \
+    && apt install -y --no-install-recommends ca-certificates gpg wget \
+    && wget -O - https://apt.kitware.com/keys/kitware-archive-latest.asc 2>/dev/null | gpg --dearmor - | tee /usr/share/keyrings/kitware-archive-keyring.gpg >/dev/null \
+    && echo 'deb [signed-by=/usr/share/keyrings/kitware-archive-keyring.gpg] https://apt.kitware.com/ubuntu/ focal main' | tee /etc/apt/sources.list.d/kitware.list >/dev/null \
+    && rm -rf /var/lib/apt/lists/*
```
Comment (asrinivasanTT) [inline tt-umd/Dockerfile:14]: To save space, move this line to the end/after all the apt commands

Comment (asrinivasanTT) [reply tt-umd/Dockerfile:14]: oh nevermind, you're already doing it

**Comment by tt-rkim** [comment]
Anything else you'd like to add? @asrinivasanTT

**Comment by asrinivasanTT** [comment]
All good

**Pull request merged**